src/Cryptol/Parser/AST.hs

```diff
@@ -985,4 +990,156 @@ instance NoPos (Prop name) where
       CLocated c _  -> noPos c
       CType t       -> CType (noPos t)
 
+-- --Typeclass for prettying up the AST for exporting to Coq
```

Comment:
* How about ShowParseable instead of ShowAST?
* Will do
* Short answer: I don't believe so. NoPos will just set all positions to (0,0), but will leave positions on the object. Further, NoPos only removes the top level of positions, and doesn't recurse on the inner structure. noPos has type t -> t, where showParseable has type t -> String. showParseable also has to print out Let as MLet (to avoid the reserved word in Coq), and perform some other small but necessary transformations in order for the definition printed out to be accepted in Coq.